Client/Application/AppDelegate.swift

```diff
@@ -86,16 +86,39 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
     }
 
     func application(application: UIApplication, openURL url: NSURL, sourceApplication: String?, annotation: AnyObject?) -> Bool {
-        if let components = NSURLComponents(URL: url, resolvingAgainstBaseURL: false) where components.scheme == "firefox" && components.host == "open-url" {
-            if let query = components.query, item = components.queryItems?.first as? NSURLQueryItem {
-                if item.name == "url" && item.value != nil {
-                    if let newURL = NSURL(string: item.value!) {
-                        let tab = self.tabManager.addTab(request: NSURLRequest(URL: newURL))
-                        self.tabManager.selectTab(tab)
+        let components = NSURLComponents(URL: url, resolvingAgainstBaseURL: false)
+        if components!.scheme != "firefox" && components!.scheme != "firefox-x-callback" {
+            return false
+        }
+        switch components!.host! {
```

Comment:
Nit: Newline above here. It helps readability to add newlines after `return`s, and also before `if`/`while`/`switch` blocks (in this case, you have both).